GVFS/GVFS.Common/Database/SparseTable.cs

```diff
@@ -68,22 +67,27 @@ public HashSet<string> GetAll()
             }
         }
 
-        public void Remove(string directory)
+        public void Remove(string directoryPath)
         {
             try
             {
                 using (IDbConnection connection = this.connectionPool.GetConnection())
                 using (IDbCommand command = connection.CreateCommand())
                 {
                     command.CommandText = "DELETE FROM Sparse WHERE path = @path;";
-                    command.AddParameter("@path", DbType.String, directory);
+                    command.AddParameter("@path", DbType.String, NormalizePath(directoryPath));
                     command.ExecuteNonQuery();
                 }
             }
             catch (Exception ex)
             {
-                throw new GVFSDatabaseException($"{nameof(SparseTable)}.{nameof(this.Remove)}({directory}) Exception: {ex.ToString()}", ex);
+                throw new GVFSDatabaseException($"{nameof(SparseTable)}.{nameof(this.Remove)}({directoryPath}) Exception: {ex.ToString()}", ex);
             }
         }
+
+        private static string NormalizePath(string path)
+        {
+            return path.Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar).Trim(Path.DirectorySeparatorChar);

```

Comment:
- Do we also need to trim whitespace?  Or will that have been done by now?
- A unit test for this method might be a good idea.

Reply:
Will add code to trim the whitespace and some functional tests.  The unit tests for this class are coming in another PR.